pkg/trace/api/api.go

```diff
@@ -762,3 +763,34 @@ func getMediaType(req *http.Request) string {
 	}
 	return mt
 }
+
+// getDirector returns a director for a httputil.ReverseProxy. It enriches requests with headers setting custom
+// container and additional tags and overwrites the user agent header.
+func getHeaderDecoratorDirector(tags string, metricSuffix string) func(req *http.Request) {
+	return func(req *http.Request) {
+		req.Header.Set("Via", fmt.Sprintf("trace-agent %s", info.Version))
+		if _, ok := req.Header["User-Agent"]; !ok {
+			// explicitly disable User-Agent so it's not set to the default value
+			// that net/http gives it: Go-http-client/1.1
+			// See https://codereview.appspot.com/7532043
+			req.Header.Set("User-Agent", "")
+		}
+		containerID := req.Header.Get(headerContainerID)
+		if ctags := getContainerTags(containerID); ctags != "" {
+			req.Header.Set("X-Datadog-Container-Tags", ctags)
+		}
+		req.Header.Set("X-Datadog-Additional-Tags", tags)
+		_ = metrics.Count("datadog.trace_agent."+metricSuffix, 1, nil, 1)
+	}
+}
+
+// getOrchestrator returns the fargate orchestrator. Should only be called if the agent is running in ECS or EKS.
```

Comment:
Well, actually, this "should only be called if ..." won't really guarantee that people won't do it. It should be ok to call this regardless and have it return `""`. The only way to achieve that is more work:
* Remove `(*AgentConfig).IsFargate` field
* Use `(pkg/util/fargate).IsFargateInstance` but create a global variable cache of the first response which stores the value to avoid executing the current function body which is an HTTP request on each call.

Or, if you don't want to do the above, then simply ignore the above along with my other comment about returning the full tag and accept that this function may be called even outside a Fargate instance and document that it will return `(pkg/util/fargate).Unknown` and handle that case in code where this is called.

Lastly, as a note, `fargate.GetOrchestrator` will involve an HTTP call too, every time. We could avoid this and only do it once at startup by replacing what is now `(*AgentConfig).IsFargate` with `(*AgentConfig).FargateOrchestrator` and just know that the value `Unknown` means we are not in Fargate. That way, this config field can account both for telling us whether we are in Fargate (like today) and also the orchestrator.